lib/request_handler/multipart_parser.rb

```diff
@@ -15,9 +15,14 @@ def initialize(request:, multipart_config:)
       raise MissingArgumentError, missing_arguments unless missing_arguments.empty?
     end
 
-    def run
+    def run # rubocop:disable AbcSize
       multipart_config.keys.each_with_object({}) do |name, memo|
-        params.fetch(name.to_s) { raise MultipartParamsError, multipart: 'missing' }
+        if multipart_config[name][:required]
+          params.fetch(name.to_s) { raise MultipartParamsError, multipart: "#{name} missing" }
```

Comment:
instead of using fetch and not really using the value I would rather suggest using

```
multipart_config.each_with_object({}) do |(name, config), memo|
  raise ... if config[:required] && !params.key?(name.to_s)
  memo[name] = parse_part(name.to_s) 
end
```